mozilla/activity-stream
# Add enable/disable/reset devtools for ASRouter

### IMPORTANT NOTE

A big part of this change is indentation so I would recommending viewing this with https://github.com/mozilla/activity-stream/pull/4508/files?w=1

### Summary
This patch adds some additional utilities to `about:newtab#asrouter`, including
* A way to enable/disable providers via the UI;
* A way to filter the messages shown by provider;
* A way to reset the preference via the UI

### Testing instructions
* Turn on devtools by setting `browser.newtabpage.activity-stream.asrouter.devtoolsEnabled` to `true`
* Visit `about:newtab#asrouter`
* Try enabling/disabling different providers; see if their messages show up/disappear on the page
* Try the messages filter dropdown under the "Messages" section
* After changing some settings, try the "Restore defaults" button and ensure `browser.newtabpage.activity-stream.asrouter.messageProviders` is the default value in `about:config`
